geode-core/src/main/java/org/apache/geode/internal/security/IntegratedSecurityService.java

```diff
@@ -221,8 +221,8 @@ public void authorize(Resource resource, Operation operation, Target target, Str
   }
 
   @Override
-  public void authorize(Resource resource, Operation operation, String target, String key) {
-    authorize(new ResourcePermission(resource, operation, target, key));
```

Comment:
eventually the Object key is going to be turned in a string to create ResourcePermission object. I would suggest leave the interface as a `String` to clearly indicate that only string form of the key will be used.

Reply:
@jinmeiliao, we wanted to defer converting the key to a string until it is actually needed. Depending on the key class's implementation of toString this conversion could be costly so should only be done if security is enabled. IntegratedSecurityService does need the key converted but LegacySecurityService (the default) does not since it never uses the key string. It sounds like you want the SecurityService interface to keep this parameter as a String. But I think that would make this optimization impossible.

Reply:
I see, this change allows you to save the call to key.toString if it's legacySecurityService. Understood now.